### ISSUE TITLE
UnifiedPush support

### ISSUE THREAD

**Issue by p1gp1g** [LeanderBB/you-have-mail-cli#2]
Hi !
I wanted to comment on the other ticket but just have been closed. It is about 2 things

1. Architecture

Just an example, [mollysocket](https://github.com/MollySocket/mollysocket) is the equivalent for signal (and [molly](https://github.com/mollyim/mollyim-android) as signal client - should be merged soon, available as a [beta](https://github.com/mollyim/mollyim-insider-android/releases)) of you-have-mail for your mail providers. You may want give a look, I've written it in rust too :)

Concerning UnifiedPush : your app (you-have-mail) will receive an endpoint with its UnifiedPush distributor (ntfy/NextPush/Conversation/etc) and this endpoint may change.

Basically, you have to set your *you-have-mail-cli* endpoint in *you-have-mail* (android) and the android app will update its endpoint when it receives a new one. (see molly/mollysocket)

2. The push message

I haven't look how you send it. But we suggest 2 solutions:
  a - A generic message and a sync-on-push strategy : the app sync when it receives the notification.
  b - Webpush : so the content of the message is encrypted. You can give a look at the [UP-Example's class related to WebPush](https://github.com/UnifiedPush/android-example/blob/main/app/src/main/java/org/unifiedpush/example/utils/WebPush.kt) to see how it works

**Comment by p1gp1g** [comment]
The current support, is nevertheless already useful, since you can use some push applications, like ntfy and NextPush to see when you have a notification :) But that is not really UnifiedPush (which is a protocol used to forward notifications to other apps)

**Comment by LeanderBB** [comment]
Hey @p1gp1g, thanks for the clarification. I misunderstood the whole ordeal :'(. I have renamed the existing implementation to avoid confusion.

I'll have to spend more time to research this. I'll definitely check out the links you provided but this does not look like a very straight forward thing. 

You are correct though, I would need to implement a backend and a notifier for this to glue together.
